commit 1862add56596942722fd7b8c205a4504805b01c5
Author: debiansid <debiansid@gmail.com>
Date:   Thu Oct 30 16:09:32 2025 +0800

    [PATCH] Add minimal manual hook stubs for min_scope_syscall_hooks

diff --git a/kernel/ksud.c b/kernel/ksud.c
index b96c5ff9..edfdfe59 100644
--- a/kernel/ksud.c
+++ b/kernel/ksud.c
@@ -57,6 +57,22 @@ bool ksu_input_hook __read_mostly = true;
 #endif
 bool ksu_execveat_hook __read_mostly = true;
 
+// === Minimal Manual Syscall Hook Stubs ===
+// Required by min_scope_syscall_hooks_v1.5.patch when CONFIG_KSU_KPROBES_HOOK=n
+
+int ksu_handle_sys_read(unsigned int fd, char __user **buf_ptr, size_t *count_ptr)
+{
+	return 0;
+}
+EXPORT_SYMBOL(ksu_handle_sys_read);
+
+int ksu_handle_input_handle_event(unsigned int *type, unsigned int *code,
+				   int *value, void *handle)
+{
+	return 0;
+}
+EXPORT_SYMBOL(ksu_handle_input_handle_event);
+
 u32 ksu_devpts_sid;
 
 void on_post_fs_data(void)
